### ISSUE TITLE
Timeout error while uploading large files over S3

### ISSUE THREAD

**Issue by holographix** [strapi/strapi#9722]
### Describe the bug

I am trying to upload video files ( 250Mb+ --> 2Gb on average ) over an S3 bucket, the problem is that while the file ( if you wait long enough, refresh the s3 file list every now and then to check ) gets correctly uploaded over the S3 bucket, the frontend part on the Strapi side instead goes in timeout after around a minute ( screenshot included ) hanging indefinitely on the Upload screen ( screenshot included ).
This topic has been already covered in the past ( issue #8772 )

### Steps to reproduce the behavior

In order to reproduce this bug simply log in your strapi admin, go to "Media Library", click the "Upload assets"  button and drag in a big file from your hard drive ( possibly a video, but we also tried with big images, such as huge tifs, with the same outcome ) and click che "Upload 1 asset to the library" button.

-- config/middleware.js

```module.exports = ({ env }) => {
  return {
    settings: {
      parser: {
        enabled: true,
        multipart: true,
        textLimit: 10737418240,
        formLimit: 10737418240,
        jsonLimit: 10737418240,
        includeUnparsed: true,
        formidable: {
          maxFileSize: 10737418240, // 10 GB
        },
      },
      cors: {
        origin: [
          "http://localhost:3000",
          "http://localhost",
          "http://mywebsite.com/",
          "https://mywebsite.com/",
          "http://mywebsite.com:1337/",
          "https://mywebsite.com:1337/",
          "http://localhost:1337",
          "http://localhost:8000",
          "http://XXX.XXX.XXX.XXX",
          "http://XXX.XXX.XXX.XXX:8000",
          "http://XXX.XXX.XXX.XXX:1337",
        ],
      },
    },
  };
};
```

-- /etc/nginx/sites-enabled/mywebsite.com.conf
```
server{
    listen 80;
    server_name mywebsite.com;
    client_max_body_size 2048M;

    location / {
        proxy_pass http://localhost:3030;
        proxy_http_version 1.1;
        proxy_set_header Upgrade $http_upgrade;
        proxy_set_header Connection 'upgrade';
        proxy_set_header Host $host;
        proxy_cache_bypass $http_upgrade;
        proxy_set_header  X-Real-IP $remote_addr;
        proxy_set_header  X-Forwarded-Proto https;
        proxy_set_header  X-Forwarded-For $remote_addr;
        proxy_set_header  X-Forwarded-Host $remote_addr;

	##
        # Upload size
        ##

        client_max_body_size 2048M;
    }
    index index.html;

    access_log /var/log/nginx/mywebsite.com.access.log;
    error_log /var/log/nginx/mywebsite.com.error.log;

}
```

### Expected behavior

It shall upload the file to S3 as expected

### Screenshots

Uploading goes on and on indefinitely 
<img width="922" alt="Strapi Uploading goes on and on indefinitely " src="https://user-images.githubusercontent.com/2369563/111139615-01f8dd80-858a-11eb-83e8-2e4cab21fcb0.png">

Timeout errors
<img width="1909" alt="Strapi upload S3 timeout errors" src="https://user-images.githubusercontent.com/2369563/111137685-c2c98d00-8587-11eb-9ee7-7d6835615177.png">

### System

- Node.js version: 12.19.0
- NPM version: 6.14.8
- Strapi version: 3.2.5
- Database: Mongo
- Operating system: OSX ( dev ) — Ubuntu 18.04 ( prod )

-- package.json dependencies extract

```
"dependencies": {
    "@ckeditor/ckeditor5-build-classic": "^23.1.0",
    "@ckeditor/ckeditor5-react": "^3.0.0",
    "strapi": "3.2.5",
    "strapi-admin": "3.2.5",
    "strapi-connector-mongoose": "3.2.5",
    "strapi-plugin-content-manager": "3.2.5",
    "strapi-plugin-content-type-builder": "3.2.5",
    "strapi-plugin-email": "3.2.5",
    "strapi-plugin-upload": "3.2.5",
    "strapi-plugin-users-permissions": "3.2.5",
    "strapi-provider-upload-aws-s3": "^3.3.4",
    "strapi-utils": "3.2.5"
  }
```

### Additional context

* I also tried to upgrade the Strapi version to the lastest ( 3.5.2 at time of writing ), but the problem doesn't seem to go away.
* It is also worth mentioning that the Strapi installation we are working on is pretty much default with no custom code whatsoever. We simply added a few Collection types to start working and that's it. 
* The screenshots have been performed on localhost, but the behaviour is exactly the same on the production environment.

**Comment by fmileto** [comment]
Same issue here, after uploading huge files the interface stucks without having visual feedbacks and i found some timeout error on console, but after a while the file is uploaded on S3 bucket

**Comment by gonzoandrea** [comment]
+1

**Comment by indieocean** [comment]
+1 Our problem could be a bit different we run in Kubernetes and every time we upload a file the Pod crashed. No logs on stdout; it just crashes. We are investigating the issue right now. We are using the s3 plugin and our pods use KIAM we can upload files from the pods without problems running `aws s3 cp`.

**Comment by derrickmehaffy** [comment]
Possibly fixed with #12099, requires testing

**Comment by bportman** [comment]
Is there any way to apply this fix to Strapi 3.6.6?  We are running an app on Mongo DB and encountering this exact issue when uploading large files through to S3.  They go into the ubuntu /tmp folder but the request times out.

**Comment by derrickmehaffy** [comment]
> Is there any way to apply this fix to Strapi 3.6.6? We are running an app on Mongo DB and encountering this exact issue when uploading large files through to S3. They go into the ubuntu /tmp folder but the request times out.

Probably not, v3 is an entirely different beast and the modifications made were targeted for the new code-base. We (Strapi) won't backport this to v3.
